### PR TITLE
Analysisd - add new test to check analysisd socket properties

### LINKED ISSUE #1850: Add a test to check that the queue socket remains after a configuration test
|Installation|Component|Version|Related issue|
|---|---|---|---|
|Manager|Analysisd|4.2.2+|https://github.com/wazuh/wazuh/issues/9985|

As reported in the issue above, we need to check that the Analysisd's input socket (_queue/sockets/queue_) file remains after running Analysisd in test mode:
```
bin/wazuh-analysisd -t
```

We can implement this test by checking that the following properties of the file won't change after running Analysisd in test mode:
- Inode
- Modification time

## Known versions where this test would fail
- 4.2.0
- 4.2.1

### PR DESCRIPTION
**Pull request by fernandolojano**
|Related issue|
|---|
|#1850|

## Description
closes #1850

This PR adds a new test for Analysisd requested on the issue #1850. As for now, it works properly as it's able to detect the changes of the file time and inode whenever
they change. 

This test check:
- The Inode value of the queue socket can't change after the Analysisd daemon is set in the testing configuration mode 
- The File time value of the queue socket can't change after the Analysisd daemon is set in the testing configuration mode 

## Tests
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.
- [x] `provision_documentation.sh` generate the docs without errors.

### CODE REVIEW

**Review by mauromalara — CHANGES_REQUESTED** [review]

**Review by mauromalara — CHANGES_REQUESTED** [review]

**Comment by mauromalara** [comment]
|CentOS: Manager|Local|Jenkins|
|---|---|---|
|PS1|[:red_circle: ](https://github.com/wazuh/wazuh-qa/files/7797129/R1-test.queue.socket.properties.mauro.tar.gz)|[:red_circle: ](https://ci.wazuh.info/job/Test_integration/18168/)
|PS2|[:red_circle: ](https://github.com/wazuh/wazuh-qa/files/7797130/R2-test.queue.socket.properties.mauro.tar.gz)|[:red_circle: ](https://ci.wazuh.info/job/Test_integration/18173/)
|PS3|[:red_circle: ](https://github.com/wazuh/wazuh-qa/files/7797131/R3-test.queue.socket.properties.mauro.tar.gz)|[:red_circle: ](https://ci.wazuh.info/job/Test_integration/18174/)

|Note|
|---|
|It fails apparently because the bug was not fixed|
|@fernandolojano please, check the results with the core team to determine if is an error or a bug of Wazuh.|

**Review by mauromalara — APPROVED** [review]

**Review by mauromalara — CHANGES_REQUESTED** [review]
@fernandolojano please, modify the code following the PEP8 guidelines. You can follow [this](https://code.visualstudio.com/docs/python/linting) guide to doing it, take into account that you must change the `max-line-length` argument following [this guide](https://code.visualstudio.com/docs/python/linting#_pycodestyle-pep8)

**Review by mauromalara — DISMISSED** [review]

**Comment by mauromalara** [comment]
## 03/02/2022
### Package
| Version | Revision | Link|
|---|---|---|
|4.4.0|0.40400.20220202|[package](https://s3.us-west-1.amazonaws.com/packages-dev.wazuh.com/staging/yum/wazuh-manager-4.4.0-0.40400.20220202.x86_64.rpm)


## Testing

### Analysisd

|  OS  | Local   | Jenkins | Notes |
|---|---|---|---|
| R1 | [🔴](https://github.com/wazuh/wazuh-qa/files/7998482/R1-1850-centos-manager.tar.gz) | [🔴](https://ci.wazuh.info/job/Test_integration/19984/artifact/manager_tests_Test_integration_B19984_20220203201525.txt) - [Build](https://ci.wazuh.info/job/Test_integration/19984/) | **Local failure:** The inode value for the socket  `/var/ossec/queue/sockets/queue` has changed -- **Jenkins failure:** The modification time property value for the socket `/var/ossec/queue/sockets/queue` has changed |
| R2 | [🔴](https://github.com/wazuh/wazuh-qa/files/7998483/R2-1850-centos-manager.tar.gz) | [🔴](https://ci.wazuh.info/job/Test_integration/19985/artifact/manager_tests_Test_integration_B19985_20220203201619.txt) - [Build](https://ci.wazuh.info/job/Test_integration/19985/) | Same as before |
| R3 | [🔴](https://github.com/wazuh/wazuh-qa/files/7998485/R3-1850-centos-manager.tar.gz) | [🔴](https://ci.wazuh.info/job/Test_integration/19986/artifact/manager_tests_Test_integration_B19986_20220203201651.txt) - [Build](https://ci.wazuh.info/job/Test_integration/19986/) | Same as before |

| NOTES (important) |
|---|
| @snaow it seems that the bug was not fixed. This may be a **stopper**. The test fails in both executions. In Jenkins, the `mtime` changes but locally the `inode` changes instead. The package used is the same. |

**Review by mauromalara — DISMISSED** [review]

**Review by jnasselle — COMMENTED** [review]
There's no stopper and this could be validated by manual testing (`ls -li` to get inodes and `stat -c '%y'` to get modification time).
I propose a rearrangement of this test in the next branch: https://github.com/wazuh/wazuh-qa/tree/fix-1850-check-queue-socket-status-test

Hope this helps!
Nico

**Comment by mauromalara** [comment]
> There's no stopper and this could be validated by manual testing (`ls -li` to get inodes and `stat -c '%y'` to get modification time). I propose a rearrangement of this test in the next branch: https://github.com/wazuh/wazuh-qa/tree/fix-1850-check-queue-socket-status-test
> 
> Jenkins: Tested using v4.2.2 staging packages Local: Tested using v4.4.0 source installation
> 
> CentOS: Manager	Local	Jenkins
> PS1	[green_circle ](https://github.com/wazuh/wazuh-qa/files/8006141/R1-test.queue.socket.properties.nico.zip)	[green_circle](https://ci.wazuh.info/job/Test_integration/20150)
> PS2	[green_circle](https://github.com/wazuh/wazuh-qa/files/8006142/R2-test.queue.socket.properties.nico.zip)	[green_circle](https://ci.wazuh.info/job/Test_integration/20151)
> PS3	[green_circle](https://github.com/wazuh/wazuh-qa/files/8006143/R3-test.queue.socket.properties.nico.zip)	[green_circle](https://ci.wazuh.info/job/Test_integration/20152)
> Hope this helps! Nico

Hi @jnasselle,

Thanks for your help, I'll check it and apply the necessary changes.

**Comment by mauromalara** [comment]
# 21/02/2022
## Package
| Version | Revision | Link|
|---|---|---|
|4.4.0|0.40400.20220202|[package](https://s3.us-west-1.amazonaws.com/packages-dev.wazuh.com/staging/yum/wazuh-manager-4.4.0-0.40400.20220202.x86_64.rpm)

## Testing

|CentOS: Manager|Local|Jenkins|
|---|---|---|
|PS1|[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8112345/R1-1850-centos-manager.zip)|[:green_circle:]()
|PS2|[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8112346/R2-1850-centos-manager.zip)|[:green_circle:]()
|PS3|[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8112347/R3-1850-centos-manager.zip)|[:green_circle:]()

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Release report: TBD
 - CVEs alerts inventory for Vulnerability Detector - VDT and WDB Integration Tests implementation ([#1243](https://github.com/wazuh/wazuh-qa/pull/1243))
 - Analysisd - add new test to check the pre-decoding stage of analysisd ([#2406](https://github.com/wazuh/wazuh-qa/pull/2406))
 - Add test to check if files can be accessed while FIM has them opened ([#705](https://github.com/wazuh/wazuh-qa/pull/705))
+- Analysisd - add new test to check analysisd socket properties ([#2405](https://github.com/wazuh/wazuh-qa/pull/2405))
```
Comment (Rebits) [inline CHANGELOG.md:17]:
```suggestion
- Analysisd - add a new test to check analysisd socket properties ([#2405](https://github.com/wazuh/wazuh-qa/pull/2405))
```

File: deps/wazuh_testing/wazuh_testing/db_interface/__init__.py [hunk 2]
```diff
@@ -22,12 +24,33 @@ def query_wdb(command):
...
+                retry += 1
+
+            # Restart wazuh-db in case of wdb socket is not yet up.
+            if not os.path.exists(WAZUH_DB_SOCKET_PATH):
+                print("Restarting wazuh-db ...")
+                control_service('restart', daemon='wazuh-db')
+
+        # Raise custom exception if the socket is not up in the expected time, even restarting wazuh-db
+        if not os.path.exists(WAZUH_DB_SOCKET_PATH):
+            raise Exception('The wdb socket is not up. wazuh-db was restarted but the socket was not found')
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/db_interface/__init__.py:45]: This logic should be performed in other fixtures or functions. This logic does not fit most of the use cases of this function so it is better to handle daemons up in test fixtures.

File: deps/wazuh_testing/wazuh_testing/db_interface/__init__.py [hunk 3]
```diff
@@ -22,12 +24,33 @@ def query_wdb(command):
...
+
+    # Create and open the socket connection with wazuh-db socket
     sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
     sock.connect(WAZUH_DB_SOCKET_PATH)
-
     data = []
 
     try:
+        # Send the query request
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/db_interface/__init__.py:53]:
```suggestion
```

File: deps/wazuh_testing/wazuh_testing/db_interface/__init__.py [hunk 4]
```diff
@@ -22,12 +24,33 @@ def query_wdb(command):
...
+            # Restart wazuh-db in case of wdb socket is not yet up.
+            if not os.path.exists(WAZUH_DB_SOCKET_PATH):
+                print("Restarting wazuh-db ...")
+                control_service('restart', daemon='wazuh-db')
+
+        # Raise custom exception if the socket is not up in the expected time, even restarting wazuh-db
+        if not os.path.exists(WAZUH_DB_SOCKET_PATH):
+            raise Exception('The wdb socket is not up. wazuh-db was restarted but the socket was not found')
+
+    # Create and open the socket connection with wazuh-db socket
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/db_interface/__init__.py:47]:
```suggestion
```

File: tests/integration/test_analysisd/test_queue_socket_properties/test_queue_socket_properties.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Variables
+
+analysisd_path = os.path.join(WAZUH_PATH, 'bin', ANALYSISD_DAEMON)
+command_exec = f'{analysisd_path} -t'
+
+# Fixtures
+
```
Comment (Rebits) [inline tests/integration/test_analysisd/test_queue_socket_properties/test_queue_socket_properties.py:70]:
```suggestion
```

File: tests/integration/test_analysisd/test_queue_socket_properties/test_queue_socket_properties.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    """Get the inode and modification time values of the 'queue' socket of 'wazuh-analysisd'"""
+    return os.stat(ANALYSISD_QUEUE_SOCKET_PATH).st_ino, os.path.getmtime(ANALYSISD_QUEUE_SOCKET_PATH)
+
+
+@pytest.fixture(scope="function")
+def restart_analysisd():
+    """Restart and wait 'wazuh-analysisd' startup"""
+    # restart analysisd daemon
+    control_service('restart', daemon='wazuh-analysisd')
+    check_daemon_status(running_condition=True, target_daemon='wazuh-analysisd')
```
Comment (Rebits) [inline tests/integration/test_analysisd/test_queue_socket_properties/test_queue_socket_properties.py:83]: Move this fixture to analysisd conftest.

File: tests/integration/test_analysisd/test_queue_socket_properties/test_queue_socket_properties.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+before_socket_properties = socket_file_properties
+after_socket_properties = socket_file_properties
+
+
+def test_queue_socket_properties(restart_analysisd, before_socket_properties, run_analysisd_test_config,
+                                 after_socket_properties):
+    '''
+    description: check if after running the configuration test of 'wazuh-analysisd' the properties
+                 of the queue socket file are changed.
```
Comment (Rebits) [inline tests/integration/test_analysisd/test_queue_socket_properties/test_queue_socket_properties.py:101]:
```suggestion
    description: Check if after running the configuration test of 'wazuh-analysisd' the properties
                 of the queue socket file are changed.
```

**Review by Rebits — CHANGES_REQUESTED** [review]
GJ, but some changes are required

**Comment by mauromalara** [comment]
## 14/03/2022
### Package
| Version | Revision | Link|
|---|---|---|
| 4.4.0 | 0.40400.20220202 | [download](https://s3.us-west-1.amazonaws.com/packages-dev.wazuh.com/staging/yum/wazuh-manager-4.4.0-0.40400.20220202.x86_64.rpm) |


## Testing

### Analysisd

|  OS  | Local   | Jenkins     | Notes
|---    |---    |---    |---    |
| R1   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8249093/R1-centos-manager-2405.tar.gz)  |  [:green_circle: ](https://ci.wazuh.info/job/Test_integration/23531/artifact/manager_tests_Test_integration_B23531_20220314213521.txt) - [Build](https://ci.wazuh.info/job/Test_integration/23531)   | |
| R2   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8249098/R2-centos-manager-2405.tar.gz)  |  [:green_circle: ](https://ci.wazuh.info/job/Test_integration/23532/artifact/manager_tests_Test_integration_B23532_20220314213654.txt) - [Build](https://ci.wazuh.info/job/Test_integration/23532/)   | |
| R3   | [:green_circle:](link-to-the-report-r3)   |  [:green_circle: ](https://ci.wazuh.info/job/Test_integration/23533/artifact/manager_tests_Test_integration_B23533_20220314213724.txt) - [Build](https://ci.wazuh.info/job/Test_integration/23533/)

**Comment by Rebits** [comment]
## 16/03/2022
### Package
| Version | Revision | Link|
|---|---|---|
|4.3.0| 1| https://packages-dev.wazuh.com/pre-release/yum/wazuh-manager-4.3.0-1.x86_64.rpm |

## Testing

### `test_analysisd`
|  OS  | Local        | Notes
|---    |---    |---  |
| R1   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8261127/R3-2405-analysisd-socket-properties.tar.gz)  | 
| R2   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8261128/R2-2405-analysisd-socket-properties.tar.gz)   | 
| R3   |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8261129/R1-2405-analysisd-socket-properties.tar.gz)   | |






* * * 

- :green_circle:: All pass
- :yellow_circle:: Some warnings
- :red_circle:: Some errors/fails
- :large_blue_circle:: In progress

**Review by Rebits — APPROVED** [review]
LGTM

**Pull request merged**